src/vs/workbench/contrib/terminal/common/capabilities/cwdDetectionCapability.ts

```diff
@@ -7,6 +7,13 @@ import { TerminalCapability } from 'vs/platform/terminal/common/terminal';
 
 export class CwdDetectionCapability {
 	readonly type = TerminalCapability.CwdDetection;
+	private _cwd = '';
+	//TODO: move shell integration addon in here
```

Comment:
I'm sure we'll need an event here that only fires if `_cwd` changes

Reply:
todo: what should listen to this